### PR TITLE
Geomtric types added

### PR DESCRIPTION
**Pull request by croudet**
Geometric types added:
     - Points
     - Line Segments
     - Boxes
     - Paths
     - Polygons
     - Circles

### CODE REVIEW

**Comment by croudet** [comment]
All test passed!

Results :

Tests run: 754, Failures: 0, Errors: 0, Skipped: 6

File: src/main/java/com/impossibl/postgres/protocol/v30/ProtocolShared.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ private void init() {
     Executor bossExecutorService = Executors.newCachedThreadPool(new NamedThreadFactory("PG-JDBC Boss"));
     Executor workerExecutorService = Executors.newCachedThreadPool(new NamedThreadFactory("PG-JDBC Worker"));
 
-    int workerCount = getRuntime().availableProcessors();
+    int workerCount = Math.max(10, getRuntime().availableProcessors());
```
Comment (kdubb) [inline src/main/java/com/impossibl/postgres/protocol/v30/ProtocolShared.java]: Please remove this change.  If anything this max should be a connection setting.

File: src/test/java/com/impossibl/postgres/jdbc/ParameterMetaDataTest.java [hunk 2]
```diff
@@ -74,7 +74,7 @@ public void testParameterMD() throws SQLException {
     assertEquals(Types.VARCHAR, pmd.getParameterType(2));
     assertEquals("text", pmd.getParameterTypeName(2));
     assertEquals("java.lang.String", pmd.getParameterClassName(2));
-    assertEquals(Types.OTHER, pmd.getParameterType(3));
+    assertEquals(Types.BINARY, pmd.getParameterType(3));
```
Comment (kdubb) [inline src/test/java/com/impossibl/postgres/jdbc/ParameterMetaDataTest.java]: What is the reason for this change?

**Comment by jesperpedersen** [comment]
Shouldn't we consider some of this for .api. ?

**Comment by jesperpedersen** [comment]
There is some code that commented out, remove that

**Comment by jesperpedersen** [comment]
Using "git rebase -i upstream/develop" to pull in upstream changes makes it easier to merge

**Comment by kdubb** [comment]
@jesperpedersen, 

@croudet is just following my lead.  I had originally put all the system data type classes (e.g. ACLItem) in .data. (for some unknown reason lol).  All of the classes in .data. should be moved to .api. at some point... I can do that once this is merged

**Comment by jesperpedersen** [comment]
@kdubb sounds good :)

File: src/test/java/com/impossibl/postgres/jdbc/ParameterMetaDataTest.java (hunk 2 above)
Comment (croudet) [inline src/test/java/com/impossibl/postgres/jdbc/ParameterMetaDataTest.java]: I am not sure what type to put for getInputPrimitiveType() and getOutputPrimitiveType(). Currently it is Binary, should it be Array?

Comment (kdubb) [inline src/test/java/com/impossibl/postgres/jdbc/ParameterMetaDataTest.java]: You can try to use Unknown.

Since these are technically "system" types, unrecognized by JDBC, you could go the long route and add an enum entry for each of them to PrimitiveType. Although, that change cascades into changes in SQLTypeMetadata, SQLTypeUtils, etc.

**Comment by kdubb** [comment]
@croudet  please do rebase as @jesperpedersen has shown when updating your PR branches.  It makes the commit history much nicer.

**Comment by jesperpedersen** [comment]
@kdubb and you shouldn't do merge commits ;) lol

**Comment by kdubb** [comment]
@jesperpedersen If I am being honest... I am no git assassin lol   I just noticed your PR's seem to come in as just the commits themselves.  I need to do some learning I know ;)

**Comment by jesperpedersen** [comment]
@kdubb see the manual instructions on the PR for starters, you can always rebase in the master repo -- the key is "git merge --ff-only"

File: src/test/java/com/impossibl/postgres/jdbc/ParameterMetaDataTest.java (hunk 2 above)
Comment (croudet) [inline src/test/java/com/impossibl/postgres/jdbc/ParameterMetaDataTest.java]: When I use Unknown I got "type has no supported parameter format" error (type.getParameterFormat().)

**Comment by croudet** [comment]
I have added the PrimitiveTypes for the geometric types.

**Comment by kdubb** [comment]
Ok so this looks pretty good.  One question I now have is,  why no types for Point, Line, Circle, etc?

Good work though!

**Comment by croudet** [comment]
I try to use types as 'raw' as possible so that the user can map them to what ever he wants.
For Path, you need to store the open/close flag so I created a class.
If you think it more convenient to have classes, I can create classes.

**Comment by kdubb** [comment]
I'm fine with it the way it is... in the future I think adding specific classes for these types would be better.

**Comment by kdubb** [comment]
Merged (by rebasing)
